### PR TITLE
Add support for https protocol for Remote drivers

### LINKED ISSUE #266: HTTPS support for remote driver
It's impossible now to provide protocol in `selenium-host` argument. 

https://github.com/pytest-dev/pytest-selenium/blob/a91b1f5e0252bb10654516f3bb73598b11a72e22/pytest_selenium/drivers/remote.py#L12 
Now `http://` hardcoded in source code, but If I need `https://` it's imposible 😞

### PR DESCRIPTION
**Pull request by platonoff-dev**
Fix #266

### CODE REVIEW

**Comment by isaulv** [comment]
Did you test with selenium `4.0.0`?

**Comment by platonoff-dev** [comment]
Yep. Looks like it work good with `4.0.0`

File: testing/test_driver.py
Comment (dosas) [inline testing/test_driver.py:110]: @platonoff-dev You added a check for https and no protocol but shouldn't you also test the case for http protocol?

File: pytest_selenium/drivers/remote.py
Comment (dosas) [inline pytest_selenium/drivers/remote.py:11]: Should we also fix this for the appium driver: https://github.com/pytest-dev/pytest-selenium/blob/a91b1f5e0252bb10654516f3bb73598b11a72e22/pytest_selenium/drivers/appium.py#L12 ?

File: pytest_selenium/drivers/remote.py
Comment (dosas) [inline pytest_selenium/drivers/remote.py:12]: @platonoff-dev I know that this is totally valid and working code but what about:
```
protocol = "" if re.search(r"^https?://", host) else "http://"
executor = f"{protocol}{host}:{port}/wd/hub
```

File: pytest_selenium/drivers/remote.py
Comment (platonoff-dev) [reply pytest_selenium/drivers/remote.py:11]: Yeah, I forgot about appium 😅 I'll do that

File: testing/test_driver.py
Comment (platonoff-dev) [reply testing/test_driver.py:110]: I think `test_default_host_port` does the same... 
https://github.com/pytest-dev/pytest-selenium/blob/a91b1f5e0252bb10654516f3bb73598b11a72e22/testing/test_driver.py#L94

Comment (dosas) [reply testing/test_driver.py:110]: No I don't think so that is another case where the protocol is not http:
host |  url   | tested |
------| ----| ------|
https://... | https://... | yes
http://... | http:// | no
no.protocol | http:// | yes
localhost | http:// | yes

File: pytest_selenium/drivers/remote.py
Comment (platonoff-dev) [reply pytest_selenium/drivers/remote.py:12]: I'm good with that. It's one of my problems 😅. I thought about that, but decided that my solution is more kind for new developers. They faster understand my version.

Comment (platonoff-dev) [reply pytest_selenium/drivers/remote.py:12]: But your suggestion looks cleaner. I'll change that

File: testing/test_driver.py
Comment (platonoff-dev) [reply testing/test_driver.py:110]: Yeah, I hurried with this a little

File: pytest_selenium/drivers/remote.py
Comment (dosas) [reply pytest_selenium/drivers/remote.py:12]: I am not sure about the regexp though ...

File: testing/test_driver.py
Comment (dosas) [reply testing/test_driver.py:110]: No problem, I think this can be solved with a simple parametrization: http, https

File: pytest_selenium/drivers/remote.py
Comment (platonoff-dev) [reply pytest_selenium/drivers/remote.py:12]:
```
❯ python test_performance.py
regexp: 0.009595889999999996
if: 0.004040118000000002
```
Almost two times slower...

Comment (platonoff-dev) [reply pytest_selenium/drivers/remote.py:12]: Code that I ran:
```python
import re
import timeit


def regexp_search():
    host = "https://some.host.com"
    port = "4444"
    protocol = "" if re.search(r"^https?://", host) else "http://"
    executor = f"{protocol}{host}:{port}/wd/hub"


def if_search():
    host = "https://some.host.com"
    port = "4444"
    executor = f"{host}:{port}/wd/hub"
    if not executor.startswith("http://") and not executor.startswith("https://"):
        executor = "http://" + executor


if __name__ == "__main__":
    print("regexp:", timeit.timeit(regexp_search, number=10000))
    print("if:", timeit.timeit(if_search, number=10000))
```

Comment (dosas) [reply pytest_selenium/drivers/remote.py:12]: Wow, thanks for testing that! I think that settles it then :D

**Comment by dosas** [comment]
@platonoff-dev Any news on this?

**Comment by platonoff-dev** [comment]
Oh, completely forgot about this again( Yeah, I'll push required changes tomorrow.

**Comment by platonoff-dev** [comment]
@dosas Fixed all review remarks
It was a hard week and only now it turned out to be done)

File: testing/test_driver.py
Comment (dosas) [inline testing/test_driver.py:134]: Since 4444 is the default there is no need to explicitly pass it here but fine for me.

**Review by dosas — APPROVED** [review]

**Comment by platonoff-dev** [comment]
Any changes I need to do?

**Comment by BeyondEvil** [comment]
Thanks for reminding me @platonoff-dev I don't think so at this point. I need to do some other work first, however. I apologise for the delay!

**Comment by BeyondEvil** [comment]
I tried to help out by rebase the branch, but I couldn't figure out how to do it. 🤷‍♂️ 

@platonoff-dev

**Comment by BeyondEvil** [comment]
Once this PR is merged, i can continue with: https://github.com/pytest-dev/pytest-selenium/pull/293

**Comment by platonoff-dev** [comment]
@BeyondEvil Ohh, I see it. But how can I do it? I have no write access, so I can't resolve it here. Just one option I can see is to rebase `master` into `https-support` and resolve conflict locally and then push. Is it ok?

**Comment by BeyondEvil** [comment]
> @BeyondEvil Ohh, I see it. But how can I do it? I have no write access, so I can't resolve it here. Just one option I can see is to rebase `master` into `https-support` and resolve conflict locally and then push. Is it ok?

Yes, that's exactly how you do it. 😊 👍 

```sh
$ git pull --rebase upstream master
```

(if you named the upstream "upstream" that is 😊 )

**Comment by BeyondEvil** [comment]
I think something went wrong with the rebase @platonoff-dev 😬

**Comment by platonoff-dev** [comment]
Yeah 😬 Give me few minutes

**Comment by platonoff-dev** [comment]
@BeyondEvil looks like black in pre-commit is broken

**Comment by BeyondEvil** [comment]
Did you run pre-commit locally?

It looks like Flake8 complains about: `testing/test_driver.py:149:1: F811 redefinition of unused 'test_host_protocol' from line 118
`

**Comment by platonoff-dev** [comment]
This one is already fixed

**Comment by BeyondEvil** [comment]
> This one is already fixed

Does pre-commit pass for you locally?

**Comment by BeyondEvil** [comment]
https://github.com/psf/black/issues/2984

**Comment by platonoff-dev** [comment]
No, to commit fix I've used `--no-verify`. I wasn't sure if I should update black version here

**Comment by BeyondEvil** [comment]
> No, to commit fix I've used `--no-verify`. I wasn't sure if I should update black version here

Sure, go ahead.

**Review by BeyondEvil — DISMISSED** [review]
LGTM!

**Comment by BeyondEvil** [comment]
Nice! Almost there!

Can you squash/rebase the all the commits to only relevant ones? (probably two, one for the https changes and one for the black bump). 🙏 🙇

**Comment by platonoff-dev** [comment]
@BeyondEvil done

**Pull request merged**
